transcom/mymove
# B 18411 show tool tip for prev approved service items and banner removal

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A882385&RoomContext=TeamRoom%3A808731&concept=TeamRoom)

## Summary

~~It was found that the UI banner that was previously agreed upon, was no longer needed. What was required was for the ToolTip to display for service items that were in a previously `APPROVED` status, so that when a service item's status is changed due to the shipment destination address being changed - the TOO can view this change.~~

The trajectory of this PR was changed by request of the gov and they wanted to be able to see the updated values that were changed for the service item when the shipment destination address was changed (which in turn, changes the destination address of destination SIT service items).

This PR does the following:
- utilizes data in the `deliveryAddressUpdate` object at the shipment level and displays that for the TOO when they go to approve the service item
- the logic that determines what displays checks `historyRecord` for the service item and shows it only if the following criteria are met: 
1. The service item was updated (in this case, the status)
2. The previous status of the service item was `APPROVED`
3. The event name of the db was `reviewShipmentAddressUpdate`
- updated and consolidated test data into single file for `ServiceItemsTable`
- renamed test data file

### How to test

Once [B-18475](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A888121&RoomContext=TeamRoom%3A808731&concept=TeamRoom) is complete & merged 
```
||
```
When you pull the code, make sure to `git merge` branch `B-18475-Service-Item-Status-Change-on-Shipment-Address-Update-Approval` so you can test locally.

1. Find a move that contains destination SIT service items
2. Using the Prime simulator, update the destination address of a shipment to trigger TOO review (I always do ZIP from 73064 -> 74133 cause I'm an Oklahoma boi)
3. Log in as the TOO and review/approve the request
4. Navigate to the MTO page and you should see the destination address change info, as well as a tag
![Screenshot 2024-01-17 at 11 23 11 AM](https://github.com/transcom/mymove/assets/136510600/0b8875a4-e02b-4b53-87eb-056a7e528f26)
